Realm/RLMObjectSchema.mm

```diff
@@ -99,7 +99,18 @@ + (instancetype)schemaForObjectClass:(Class)objectClass {
     schema.properties = props;
 
     // verify that we didn't add any properties twice due to inheritance
-    assert(props.count == [NSSet setWithArray:[props valueForKey:@"name"]].count);
+    if (props.count != [NSSet setWithArray:[props valueForKey:@"name"]].count) {
```

Comment:
this could also be `[props valueForKeyPath:@"distinctUnionOfObjects.name"].count`, if you wanted to get cheeky